yt/utilities/file_handler.py

```diff
@@ -111,6 +111,6 @@ def __init__(self, filename):
     def open_ds(self):
         from yt.utilities.on_demand_imports import _netCDF4 as netCDF4
 
-        ds = netCDF4.Dataset(self.filename)
+        ds = netCDF4.Dataset(self.filename, "r")
```

Comment:
wild guessing here. If I'm right, please commit this. If I'm wrong, it shows that positional arguments are not clear enough, please add the argument name explicitly
```suggestion
        ds = netCDF4.Dataset(self.filename, mode="r")
```

Reply:
latest commit uses the `mode` keyword.